### ISSUE TITLE
Bug with tag csa_guzzle.client

### ISSUE THREAD

**Issue by kassko** [csarrazi/CsaGuzzleBundle#47]
Hello @csarrazi 

First of all,

I read the documentation for the migration to `1.3` which tells us to register clients in the following way:

```xml
<service id="acme.client" class="%acme.client.class%">
    <tag name="csa_guzzle.client"/>
</service>
```
But it seems to be useless to add the tag `csa_guzzle.client` if we do not have subscribers to tie to our client, so the following config seems to be right:
```xml
<service id="acme.client" class="%acme.client.class%"/>
```

And only if we have a subscriber to tie, we need to use this tag `csa_guzzle.client`:
```xml
<service id="acme.client" class="%acme.client.class%">
    <tag name="csa_guzzle.client" subscribers="some_subscriber"/>
</service>
```

Am I wrong ?

**Comment by kassko** [comment]
There is a bug in Symfony, with configurators that configure private services.

When used with subscribers, this tag doesn't work, we got an error in `Symfony\Component\DependencyInjection\Dumper\XmlDumper`.

We can fix this bug by turning `public` the configurator service created on the fly in the pass SubscriberPass. Just by removing the following line:

```php
$configurator->setPublic(false);
```

**Comment by csarrazi** [comment]
Hi @kassko, and thank you for your feedback!

Regarding the `csa_guzzle.client` tag, actually, if you don't specify any subscriber in the subscribers key, all subscribers will be registered in your client. This allows for easy configuration, and compatibility with other bundles.

We could improve this by using a separate tag, but this won't be the case before `2.0` (which should be released soon after we receive feedback from https://github.com/guzzle/guzzle-services/issues/89).

Regarding the configurator bug, could you provide:

* a stack trace
* the symfony version that you are using
* link to an issue in `symfony/symfony` (if there is one)

This would be great feedback for solving this bug! :)

Regardless, until it is fixed in `symfony/symfony`, we can rely on your suggestion, and make the service public.

**Comment by kassko** [comment]
Regarding the `csa_guzzle.client`, ok I understand better this tag  but effectively it's not intuituive that specifying it add automatically all subscribers.

Regarding the configurator bug, this bug is played with `Symfony 2.5`.

In all services definitions, when possible, symfony/dependency-injection makes arguments of type service in line. And so replaces the references by their real definitions (Compiler\InlineServiceDefinitionsPass::inlineArguments()). 

But the XmlDumper does not handle the case of arguments containing a real service definition instead of a service reference (Dumper\XmlDumper::addService()).

This bug is fixed in version 2.7 
https://github.com/symfony/DependencyInjection/commit/0a81a1fd6dd0d00e042f7c5ef26eff93715c6728 

However, warning ! I noticed that some SF 2.5 projects work fine with guzzle-bundle v1.3 (that can explain why there was no issue yet) so guzzle-bundle v1.3 doesn't really depend on SF 2.7 to work fine and then this issue doesn't contest the actually composer requirements `"symfony/framework-bundle": "~2.3|~3.0"`.

If anyone has the bug, the solution is to upgrade to SF 2.7 ! Anyway SF 2.5 is not maintained anymore, it is wrong to still use this version.

Thanks for your reactivity !
